lib/developer_portal/lib/liquid/drops/provider.rb

```diff
@@ -215,7 +215,16 @@ def logo_url
         </ul>
       )
       def api_specs
-        Drops::Collection.for_drop(Drops::ApiSpec).new(@model.api_docs_services.accessible)
+        service_api_specs = @model.all_api_docs.accessible(::Service)
+        account_api_specs = @model.all_api_docs.accessible(::Account)
+        backend_api_api_specs = @model.all_api_docs.accessible(::BackendApi)
+        accessible_api_specs = ApiDocs::Service.where(id: service_api_specs | account_api_specs | backend_api_api_specs)
```

Comment:
I think the logic for this should remain inside the scope. Better if we can trust `ApiDocs::Service.accessible` returns api_docs whose owners are not deleted/marked for deletion.